site/docs/extensions/index.md

```diff
@@ -0,0 +1,38 @@
+# Extensions
+
+In many cases, the existing objects in Substrait will be sufficient to accomplish a particular use case. However, it is sometimes helpful to create a new data type, scalar function signature or some other custom representation within a system. For that, Substrait provides a number of extension points. 
+
+## Simple Extensions
+
+Some kinds of primitives are so frequently extended that Substrait defines a standard YAML format that describes how the extended functionality can be interpreted. This allows different projects/systems to use the YAML definition as a specification to interoperate around the extended items so that interoperability isn't constrained solely the base Substrait specification. The main types of extensions that are defined in this manner include the following:
+
+* Data types
+* Type variations
+* Scalar Functions
+* Aggregate Functions
+* Window Functions
+* Table Functions
+
+To extend these items, developers can create one or more YAML files at a defined URI that describes the properties of each of these extensions. The YAML file is constructed according to the [YAML Schema](https://github.com/substrait-io/substrait/blob/main/text/simple_extensions_schema.yaml). Each definition in the file corresponds to the yaml based serialization of the relevant data structure. If a user only wants to extend one of these types of objects (e.g. types), a developer does not have to provide definitions for the other extension points.
+
+A Substrait plan can reference one or more YAML files via URI for extension. In the places where these entities are referenced, they will be referenced using a URI + name reference. The name scheme per type works as follows:
+
+| Category            | Naming scheme                                                |
+| ------------------- | ------------------------------------------------------------ |
+| Types               | The name as defined on the type object.                      |
+| Type Variations     | The name as defined on the type object.                      |
+| Function Signatures | If there is only one impl, the name may be defined at the top-level definition of the object. If there are multiple impls, the name is a composite string that includes the top-level name (e.g. add) followed by a compound string that describes the types of the arguments (e.g. add:i8_i8). In the case there is one impl, the multiple impl naming scheme is also supported. How arguments are mapped to a compound name are described below. |
```

Comment:
> How arguments are mapped to a compound name are described below.

??

Reply:
Yeah... I was thinking of doing a follow-on patch with that detail. I was little concerned we might get wrapped around the axle about the specifics of that which would hold up the rest of this patch. I'll add a TBD here.

Reply:
I added this to updated PR. Hopefully we won't get too caught up.